pyodide_build/build_env.py

```diff
@@ -334,10 +335,16 @@ def _create_constraints_file() -> str:
     if not constraints:
         return ""
 
-    if len(constraints.split(maxsplit=1)) > 1:
-        raise ValueError(
-            "PIP_CONSTRAINT contains spaces so pip will misinterpret it. Make sure the path to pyodide has no spaces.\n"
-            "See https://github.com/pypa/pip/issues/13283"
+    # If a path to a file specified PIP_CONSTRAINT contains spaces, pip will misinterpret
+    # it as multiple files; see https://github.com/pypa/pip/issues/13283. This is fine if
+    # the user wants to use multiple constraints files, but if they use a single file, we
+    # warn them about it and ask to convert the path to a URI or to remove spaces in it.
+    if " " in constraints:
+        logger.info(
+            "The value of PIP_CONSTRAINT contains spaces, and pip will interpret it as "
+            "multiple files. If you are using a single constraints file and it has spaces "
+            "in its file path, please convert it to a URI instead. Please ignore this "
+            "message if you are using multiple constraints files."
         )
 
     constraints_file = Path(constraints)
```

Comment:
I suppose you'd have to split and take the first entry if you're keeping the following code? `constraints` might contain two files, so `Path(constraints)` wouldn't refer to a file.